pkg/azuredisk/controllerserver.go

```diff
@@ -145,6 +149,12 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 			diskEncryptionSetID = v
 		case tagsField:
 			customTags = v
+		case pvNameKeyField:
```

Comment:
these values should be from k8s other than user input, as I could see in below logs, `kubernetes.io-created-for-pv-name` could be `req.GetName()`, I did not see we could get pvc name from `CreateVolume` now, let's add `kubernetes.io-created-for-pv-name`

```
I1225 14:40:04.687393       1 utils.go:84] GRPC call: /csi.v1.Controller/CreateVolume
I1225 14:40:04.687413       1 utils.go:85] GRPC request: {"accessibility_requirements":{"preferred":[{"segments":{"topology.disk.csi.azure.com/zone":""}}],"requisite":[{"segments":{"topology.disk.csi.azure.com/zone":""}}]},"capacity_range":{"required_bytes":10737418240},"name":"pvc-2eccae3e-2f51-43d1-ae03-9bfb87fe5135","parameters":{"skuName":"Standard_LRS"},"volume_capabilities":[{"AccessType":{"Mount":{"fs_type":"ext4","mount_flags":["barrier=1","acl"]}},"access_mode":{"mode":1}}]}
```

Reply:
The idea behind my changes is that if you add the --extra-create-metadata tag in the arguments for csi-provisioner in the csi-azuredisk-controller.yaml, it would pass those values to CreateVolume. Is this what you mean by user input?

Reply:
those tags are pv, pvc names, it's dynamic, we cannot input those names in storage class since one storage class could provision hundreds of pv/pvc, I have created a PR to add `kubernetes.io-created-for-pv-name`: https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/649, looks like there is no way to get `kubernetes.io-created-for-pvc-name`, `kubernetes.io-created-for-pvc-namespace` in CSI CreateVolume interface.